crates/adapterlib/src/errors/controller.rs

```diff
@@ -588,6 +588,7 @@ impl ResponseError for ControllerError {
             Self::Config { .. } => StatusCode::BAD_REQUEST,
             Self::UnknownInputEndpoint { .. } => StatusCode::NOT_FOUND,
             Self::ParseError { .. } => StatusCode::BAD_REQUEST,
+            Self::NotSupported { .. } => StatusCode::BAD_REQUEST,
```

Comment:
maybe this is better

[6.5.5](https://datatracker.ietf.org/doc/html/rfc7231#section-6.5.5).  405 Method Not Allowed

   The 405 (Method Not Allowed) status code indicates that the method
   received in the request-line is known by the origin server but not
   supported by the target resource.  The origin server MUST generate an
   Allow header field in a 405 response containing a list of the target
   resource's currently supported methods.

Reply:
I don't mind changing it if you think it's better.